test1103_2.rb

```diff
@@ -1,41 +1,27 @@
-def pushes(l,a)
-    l.append(a)
-end
-
-def popes(l)
-    if l.length!=0
-        l.pop
-    end
-end
-
-def topes(l)
-    if l.length!=0
-        return l[l.length-1]
-    end
-end
+require_relative 'stack_implement.rb'
 
 def checks(l,a)
     for i in 0..a.length-1
         if a[i]=='{' || a[i]=='[' || a[i]=='('
-            pushes(l,a[i])
+            l.push_stack(a[i])
         else
-            if (a[i]==')' and topes(l)=='(') || (a[i]=='}' and topes(l)=='{') || (a[i]==']' and topes(l)=='[')
-                popes(l)
+            if (a[i]==')' and l.top_of_stack=='(') || (a[i]=='}' and l.top_of_stack=='{') || (a[i]==']' and l.top_of_stack=='[')
+                l.pop_stack
             elsif a[i]==')' or a[i]=='}' or a[i]==']'
                 return false
             end
         end
     end
-    if l.length!=0
+    if l.size_of_stack!=0
         return false
     end
     return true
 end
 
-l=[]
+stack_object=Stackss.new
 puts "Enter the brackets string input here : "
-a=gets.chomp()
-if checks(l,a)
+bracket_input=gets.chomp()
```

Comment:
Get the input from the file, instead from CLI.
Assume a .txt file is given as input. Each line has one equation